streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -325,6 +334,12 @@ public void checkpoint(final Map<TopicPartition, Long> ackedOffsets) {
         }
         // write the checkpoint file before closing, to indicate clean shutdown
         try {
+            if (checkpoint == null) {
+                assert eosEnabled;
+                checkpoint = new OffsetCheckpoint(new File(baseDir, CHECKPOINT_FILE_NAME));
+            } else {
+                assert !eosEnabled;
```

Comment:
We don't use asserts because they are disabled by default unless a JVM param is passed. And they don't provide much of a benefit over just doing it via library methods.